src/sync/server.rs

```diff
@@ -473,38 +485,68 @@ impl Server {
 
                 // notify reaper thread to exit.
                 drop(reaper_tx);
-                reaper.join().unwrap();
                 info!("ttrpc server stopped");
             })
             .unwrap();
 
         self.handler = Some(handler);
-
+        info!("server listen started");
         Ok(())
     }
 
-    pub fn shutdown(mut self) {
-        let connections = self.connections.lock().unwrap();
+    pub fn start(&mut self) -> Result<()> {
+        if self.thread_count_default >= self.thread_count_max {
+            return Err(Error::Others(
+                "thread_count_default should smaller than thread_count_max".to_string(),
+            ));
+        }
+        if self.thread_count_default <= self.thread_count_min {
+            return Err(Error::Others(
+                "thread_count_default should biger than thread_count_min".to_string(),
+            ));
+        }
+        self.start_listen()?;
+        info!("server started");
+        Ok(())
+    }
 
+    pub fn shutdown_listen(mut self) -> Self {
         self.quit.store(true, Ordering::SeqCst);
         close(self.monitor_fd.1).unwrap_or_else(|e| {
             warn!(
                 "failed to close notify fd: {} with error: {}",
                 self.monitor_fd.1, e
             )
         });
+        info!("close monitor");
+        if let Some(handler) = self.handler.take() {
+            handler.join().unwrap();
+        }
+        info!("join listen");
+        self
+    }
+
+    pub fn shutdown_connection(mut self) {
+        info!("shutdown connection begin");
```

Comment:
Is `start to shutdown connection` better?

Reply:
fixed